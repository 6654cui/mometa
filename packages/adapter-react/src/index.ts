/** *
react 适配器，包括 editor compile / editor render
* @author
余聪
*/

export default function adapterReact() {}
