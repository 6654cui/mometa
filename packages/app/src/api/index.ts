import axiosStatic from 'axios'

const axios = axiosStatic.create({
  baseURL: 'https://jsonplaceholder.typicode.com/'
})

export async function getUsers() {
  const res = await axios.get('/users')
  return res.data
}
