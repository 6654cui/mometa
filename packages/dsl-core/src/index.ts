/**
 * DSL 的 parser &amp; stringify
 * @author 余聪
 */
import { ModuleNode } from './type'
import { parse, parseExpression } from '@babel/parser'

interface CommonOptions {
  plugins?: any[]
}
interface ParseOptions extends CommonOptions {
  id?: ModuleNode['id']
}
interface StringifyOptions extends CommonOptions {}

export function parseModuleNode(moduleCode: string, { id = null }: ParseOptions = {}): ModuleNode {
  return {
    type: 'module',
    id,
    imports: [],
    body: [],
    exports: []
  }
}

export function stringifyModuleNode(moduleNode: ModuleNode, opts?: StringifyOptions): { code: string } {
  return { code: '' }
}
