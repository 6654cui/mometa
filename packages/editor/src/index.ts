/** *
编辑器
* @author
余聪
*/

export default function editor() {}
