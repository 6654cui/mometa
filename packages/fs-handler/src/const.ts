export const enum OpType {
  DEL,
  REPLACE_NODE,
  MOVE_NODE,
  INSERT_NODE
}
