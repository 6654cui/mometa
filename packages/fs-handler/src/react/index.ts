import { Middleware } from '../index'
import reactDelMiddleware from './del'
import reactReplaceNodeMiddleware from './replace-node'

export default function reactMiddlewares(): Middleware[] {
  return [reactDelMiddleware(), reactReplaceNodeMiddleware()]
}
