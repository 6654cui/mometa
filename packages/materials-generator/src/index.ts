import Module = NodeJS.Module

export * from './utils/resolve-async-config'
export * from './utils/search-core'

export * from './plugins/create'

import { assets } from './plugins/assets'
import { groups } from './plugins/groups'
import { materials } from './plugins/materials'

export { assets, groups, materials }

// @ts-ignore
type SearchFn<T> = (dir: Parameters<T>[0]) => ReturnType<T>

export function createSearchHelper(mod: Module) {
  return {
    materials: ((dirs) => materials(dirs, mod.path)) as SearchFn<typeof materials>,
    groups: ((dirs) => groups(dirs, mod.path)) as SearchFn<typeof groups>,
    assets: ((dirs) => assets(dirs, mod.path)) as SearchFn<typeof assets>
  }
}
