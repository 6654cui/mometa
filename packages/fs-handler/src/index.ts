/** *
 file updator
 * @author
 余聪
 */
import { waterFall } from 'run-seq'
import pify from 'pify'
import lazy from 'lazy-value'
import { createLineContentsByContent, LineContents, Range } from './utils/line-contents'
import { OpType } from './const'
export * from './const'

export { default as reactMiddlewares } from './react'

export interface CommonPreload extends Range {
  filename: string
  name: string
  text: string
}

export interface DelPreload extends CommonPreload {}

export interface RequestData {
  type: OpType.DEL
  preload: DelPreload
}

type Fs = Pick<typeof import('fs'), 'readFile' | 'writeFile'>

export interface MiddlewareContext {
  fs: Fs
  filename: string
  getContent: () => Promise<string>

  assertLocateByRange: () => Promise<{
    lineContents: LineContents
    lines: ReturnType<LineContents['locateByRange']>
  }>

  writeFile: (data: any) => Promise<void>
}

export type Middleware = (request: RequestData, ctx: MiddlewareContext, next: () => any) => any

const apiMiddle: Middleware = (data, ctx, next) => {
  ctx.assertLocateByRange = async () => {
    const content = await ctx.getContent()
    const lineContents = createLineContentsByContent(content, { filename: ctx.filename })
    const lines = lineContents.locateByRange(data.preload)

    const string = new LineContents(
      lines.map((x) => x.line),
      { filename: ctx.filename }
    ).toString(false)

    if (string !== data.preload.text) {
      throw new Error(`匹配错误
${JSON.stringify(string)} !== ${JSON.stringify(data.preload.text)}`)
    }
    return {
      lineContents,
      lines
    }
  }

  ctx.writeFile = async (data) => {
    return pify(ctx.fs.writeFile)(ctx.filename, data)
  }

  return next()
}

export function createFsHandler({ fs, middlewares = [] }: { fs: Fs; middlewares?: Middleware[] }) {
  return async (request: RequestData) => {
    // @ts-ignore
    const ctx: MiddlewareContext = {
      fs,
      filename: request.preload.filename,
      getContent: lazy(() => pify(fs.readFile)(request.preload.filename, 'utf8'))
    }

    return await waterFall([apiMiddle].concat(middlewares), [request, ctx])
  }
}
