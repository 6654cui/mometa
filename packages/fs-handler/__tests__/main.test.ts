import { createFsHandler, OpType, commonMiddlewares } from '../src'
import * as fs from 'fs'
import { fixture } from './helper'

const mockFs = {
  readFile: fs.readFile,
  writeFile: fs.writeFile
}

describe('fsHandler', function () {
  const raw = mockFs.writeFile
  let result
  let fsHandler
  beforeEach(() => {
    fsHandler = createFsHandler({
      fs: mockFs,
      middlewares: commonMiddlewares()
    })
    // @ts-ignore
    Object.defineProperty(mockFs, 'writeFile', {
      value: (path, data, cb) => {
        result = data
        cb(null)
      },
      configurable: true
    })
  })
  afterEach(() => {
    mockFs.writeFile = raw
    result = null
  })

  it('spec case DEL', async function () {
    await fsHandler({
      type: OpType.DEL,
      preload: {
        name: 'p',
        text: '<p>单独 p</p>',
        filename: fixture('react-comp.tsx'),
        start: { line: 18, column: 10 },
        end: { line: 18, column: 21 }
      }
    })
    expect(result).toMatchSnapshot()
  })

  it('spec case REPLACE_NODE', async function () {
    await fsHandler({
      type: OpType.REPLACE_NODE,
      preload: {
        data: {
          newText: '<span>ppp</span>'
        },
        name: 'p',
        text: '<p>单独 p</p>',
        filename: fixture('react-comp.tsx'),
        start: { line: 18, column: 10 },
        end: { line: 18, column: 21 }
      }
    })
    expect(result).toMatchSnapshot()
  })

  it('nest REPLACE_NODE', async function () {
    await fsHandler({
      type: OpType.REPLACE_NODE,
      preload: {
        start: { line: 17, column: 10 },
        end: { line: 20, column: 14 },
        filename: fixture('nested.tsx'),
        name: 'p',
        text: '<p>\n            nested\n            <strong>hahahax</strong>\n          </p>',
        data: {
          newText: '<p>\n            nested 2\n            <strong>hahahax</strong>\n          </p>'
        }
      }
    })
    expect(result).toMatchSnapshot()
  })

  it('MOVE_NODE', async function () {
    await fsHandler({
      type: OpType.MOVE_NODE,
      preload: {
        text: '<p>单独 p</p>',
        filename: fixture('react-comp.tsx'),
        start: { line: 18, column: 10 },
        end: { line: 18, column: 21 },
        data: {
          to: {
            line: 17,
            column: 10
          }
        }
      }
    })
    expect(result).toMatchSnapshot()
  })

  it('INSERT_NODE', async function () {
    await fsHandler({
      type: OpType.INSERT_NODE,
      preload: {
        text: '<p>单独 p</p>',
        filename: fixture('react-comp.tsx'),
        to: { line: 18, column: 10 },
        data: {
          newText: '<p className="empty-2"></p>'
        }
      }
    })
    expect(result).toMatchSnapshot()
  })
})
