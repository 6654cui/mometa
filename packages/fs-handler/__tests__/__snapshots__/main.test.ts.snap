// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fsHandler INSERT_NODE 1`] = `
"import React, { StrictMode } from 'react'
import Tabs from 'antd/es/tabs'
import 'antd/es/tabs/style/index.css'

type Props = {
  x: string
}

const array = new Array(100).fill(1)

export default function App(props: Props) {
  return (
    <div>
      <h1 title={'abc'}>Hello World</h1>
      <Tabs>
        <Tabs.TabPane key={'tool'} tab={'Áâ©Êñô'}>
          <p className=\\"empty\\"></p>
          <p className=\\"empty-2\\"></p>
          <p>ÂçïÁã¨ p</p>
          {array.map((x, i) => (
            <p key={i}>Áâ©Êñô__{i}</p>
          ))}
        </Tabs.TabPane>
        <Tabs.TabPane key={'attr'} tab={'Â±ûÊÄß'}></Tabs.TabPane>
      </Tabs>
    </div>
  )
}"
`;

exports[`fsHandler INSERT_NODE wrap 1`] = `
"import React, { StrictMode } from 'react'
import Tabs from 'antd/es/tabs'
import 'antd/es/tabs/style/index.css'

type Props = {
  x: string
}

const array = new Array(100).fill(1)

export default function App(props: Props) {
  return (
    <div>
      <h1 title={'abc'}>Hello World</h1>
      <Tabs>
        <Tabs.TabPane key={'tool'} tab={'Áâ©Êñô'}>
          <>
            <p className=\\"empty\\"></p>
            <p className=\\"empty-2\\"></p>
          </>
          <p>ÂçïÁã¨ p</p>
          {array.map((x, i) => (
            <p key={i}>Áâ©Êñô__{i}</p>
          ))}
        </Tabs.TabPane>
        <Tabs.TabPane key={'attr'} tab={'Â±ûÊÄß'}></Tabs.TabPane>
      </Tabs>
    </div>
  )
}"
`;

exports[`fsHandler INSERT_NODE wrap 2`] = `
"import React, { StrictMode } from 'react'
import Tabs from 'antd/es/tabs'
import 'antd/es/tabs/style/index.css'

type Props = {
  x: string
}

const array = new Array(100).fill(1)

export default function App(props: Props) {
  return (
    <div>
      <h1 title={'abc'}>Hello World</h1>
      <Tabs>
        <Tabs.TabPane key={'tool'} tab={'Áâ©Êñô'}>
          <>
            <p className=\\"empty-2\\"></p>
            <p className=\\"empty\\"></p>
          </>
          <p>ÂçïÁã¨ p</p>
          {array.map((x, i) => (
            <p key={i}>Áâ©Êñô__{i}</p>
          ))}
        </Tabs.TabPane>
        <Tabs.TabPane key={'attr'} tab={'Â±ûÊÄß'}></Tabs.TabPane>
      </Tabs>
    </div>
  )
}"
`;

exports[`fsHandler INSERT_NODE wrap 3`] = `
"import React, { StrictMode } from 'react'
import Tabs from 'antd/es/tabs'
import 'antd/es/tabs/style/index.css'

type Props = {
  x: string
}

const array = new Array(100).fill(1)

export default function App(props: Props) {
  return (
    <div>
      <h1 title={'abc'}>Hello World</h1>
      <Tabs>
        <Tabs.TabPane key={'tool'} tab={'Áâ©Êñô'}>
          <p className=\\"empty\\"></p>
          <>
            <p className=\\"empty-2\\"></p>
          </>
          <p>ÂçïÁã¨ p</p>
          {array.map((x, i) => (
            <p key={i}>Áâ©Êñô__{i}</p>
          ))}
        </Tabs.TabPane>
        <Tabs.TabPane key={'attr'} tab={'Â±ûÊÄß'}></Tabs.TabPane>
      </Tabs>
    </div>
  )
}"
`;

exports[`fsHandler MOVE_NODE 1`] = `
"import React, { StrictMode } from 'react'
import Tabs from 'antd/es/tabs'
import 'antd/es/tabs/style/index.css'

type Props = {
  x: string
}

const array = new Array(100).fill(1)

export default function App(props: Props) {
  return (
    <div>
      <h1 title={'abc'}>Hello World</h1>
      <Tabs>
        <Tabs.TabPane key={'tool'} tab={'Áâ©Êñô'}>
          <p>ÂçïÁã¨ p</p>
          <p className=\\"empty\\"></p>
          {array.map((x, i) => (
            <p key={i}>Áâ©Êñô__{i}</p>
          ))}
        </Tabs.TabPane>
        <Tabs.TabPane key={'attr'} tab={'Â±ûÊÄß'}></Tabs.TabPane>
      </Tabs>
    </div>
  )
}"
`;

exports[`fsHandler getAddMaterialOps fsHandler 1`] = `
"import React, { StrictMode } from 'react'
import Tabs from 'antd/es/tabs'
import 'antd/es/tabs/style/index.css'
import Antd from 'antd'
import { Button } from 'antd'
import 'antd/lib/button/style/css'
import 'antd/lib/button/style'

type Props = {
  x: string
}

const array = new Array(100).fill(1)

export default function App(props: Props) {
  return (
    <div>
      <h1 title={'abc'}>Hello World</h1>
      <Tabs>
        <Tabs.TabPane key={'tool'} tab={'Áâ©Êñô'}>
          <p className=\\"empty\\"></p>
          <>
            <>
              <Antd />
              <Button type=\\"default\\">ÊåâÈíÆ</Button>
            </>
          </>
          <p>ÂçïÁã¨ p</p>
          {array.map((x, i) => (
            <p key={i}>Áâ©Êñô__{i}</p>
          ))}
        </Tabs.TabPane>
        <Tabs.TabPane key={'attr'} tab={'Â±ûÊÄß'}></Tabs.TabPane>
      </Tabs>
    </div>
  )
}"
`;

exports[`fsHandler nest REPLACE_NODE 1`] = `
"import React, { StrictMode } from 'react'
import Tabs from 'antd/es/tabs'
import 'antd/es/tabs/style/index.css'

type Props = {}

const array = new Array(100).fill(1)

export default function App(props: Props) {
  return (
    <div>
      <h1 title={'abc'}>Hello Worldüëå</h1>
      <Tabs>
        <Tabs.TabPane key={'tool'} tab={'Áâ©Êñô'}>
          <p className=\\"empty\\"></p>
          <p>simple 66xxxxxasdasdas6</p>
          <p>
            nested 2<strong>hahahax</strong>
          </p>
          {array.map((x, i) => (
            <p key={i}>Áâ©Êñô_{i}</p>
          ))}
        </Tabs.TabPane>
        <Tabs.TabPane key={'attr'} tab={'Â±ûÊÄß'}></Tabs.TabPane>
      </Tabs>
    </div>
  )
}"
`;

exports[`fsHandler spec case DEL 1`] = `
"import React, { StrictMode } from 'react'
import Tabs from 'antd/es/tabs'
import 'antd/es/tabs/style/index.css'

type Props = {
  x: string
}

const array = new Array(100).fill(1)

export default function App(props: Props) {
  return (
    <div>
      <h1 title={'abc'}>Hello World</h1>
      <Tabs>
        <Tabs.TabPane key={'tool'} tab={'Áâ©Êñô'}>
          <p className=\\"empty\\"></p>
          {array.map((x, i) => (
            <p key={i}>Áâ©Êñô__{i}</p>
          ))}
        </Tabs.TabPane>
        <Tabs.TabPane key={'attr'} tab={'Â±ûÊÄß'}></Tabs.TabPane>
      </Tabs>
    </div>
  )
}"
`;

exports[`fsHandler spec case REPLACE_NODE 1`] = `
"import React, { StrictMode } from 'react'
import Tabs from 'antd/es/tabs'
import 'antd/es/tabs/style/index.css'

type Props = {
  x: string
}

const array = new Array(100).fill(1)

export default function App(props: Props) {
  return (
    <div>
      <h1 title={'abc'}>Hello World</h1>
      <Tabs>
        <Tabs.TabPane key={'tool'} tab={'Áâ©Êñô'}>
          <p className=\\"empty\\"></p>
          <span>ppp</span>
          {array.map((x, i) => (
            <p key={i}>Áâ©Êñô__{i}</p>
          ))}
        </Tabs.TabPane>
        <Tabs.TabPane key={'attr'} tab={'Â±ûÊÄß'}></Tabs.TabPane>
      </Tabs>
    </div>
  )
}"
`;
